docs/content/docs/deployment/memory/network_mem_tuning.md

```diff
@@ -0,0 +1,187 @@
+of the buffers. However ideal values for those numbers are hard to pick, as they are different for every
+deployment. The buffer debloating mechanism added in Flink 1.14 attempts to address this issue.
+It tries to automatically adjust the amount of in-flight data to reasonable values.
+More precisely, the buffer debloating calculate the maximum possible throughput
+(the maximum throughput which would be if the subtask was always busy)
+for the subtask and adjusts the amount of in-flight data in such a way that the time for consumption of those in-flight data will be equal to the configured value.
+
+The most important settings:
+* The buffer debloat can be enabled by setting the property `taskmanager.network.memory.buffer-debloat.enabled` to `true`. 
+* The targeted time to consume the in-flight data can be configured by setting `taskmanager.network.memory.buffer-debloat.target` to `duration`.
+  The default value of the debloat target should be good enough in most cases.
+
+Buffer debloating in Flink works by measuring past throguhput to predict future time to consume the remaining
+in-flight data. If those predictions are incorrect, the debloating mechanism can fail in one of the two ways:
+* There won't be enough buffered data to provide full throughput
+* There will be too many buffered in-flight data and the aligned checkpoint barriers propagation time or the unaligned checkpoint size will suffer.
+
+Hence, if you have a varying load in your job, for example a sudden spikes of incoming records, or periodically
+firing windowed aggregations or joins, you might need to adjust the following settings:
+
+* `taskmanager.network.memory.buffer-debloat.period` - The minimum time between buffer size recalculation.
+The shorter the period, the faster reaction time of the debloating mechanism, but a higher CPU overhead for the necessary calculations.
+* `taskmanager.network.memory.buffer-debloat.samples` - Adjust the number of samples over which throughput measurements are averaged out.
+The frequency of the collected samples can be adjusted via `taskmanager.network.memory.buffer-debloat.period`.
+The fewer samples, the faster reaction time of the debloating mechanism, but a higher chance of a sudden spike or drop of the throughput to cause the buffer debloating to miscalculate the best amount of the in-flight data.
+* `taskmanager.network.memory.buffer-debloat.threshold-percentages` - The optimization which prevents 
+the frequent buffer size change if the new size is not so different compared to the old one.
+
+See the [Configuration]({{< ref "docs/deployment/config" >}}#full-taskmanageroptions) documentation for details and additional parameters.
+
+The [metrics]({{< ref "docs/ops/metrics" >}}#io) which can help to observe the current buffer size:
+* `estimatedTimeToConsumerBuffersMs` - the total time to consume data from all input channels.
+* `debloatedBufferSize` - the current buffer size.
```

Comment:
```suggestion
Previously, the only way to configure the amount of in-flight data was to specify both the amount and the buffer size. However, ideal values can be difficult to choose since they are different for every
deployment. The buffer debloating mechanism added in Flink 1.14 attempts to address this issue by automatically adjusting the amount of in-flight data to reasonable values.

The buffer debloating feature calculates the maximum possible throughput for the subtask (in the scenario that it is always busy) and adjusts the amount of in-flight data such that the consumption time of those in-flight data will be equal to the configured value.

The buffer debloat mechanism can be enabled by setting the property `taskmanager.network.memory.buffer-debloat.enabled` to `true`. 
The targeted time to consume the in-flight data can be configured by setting `taskmanager.network.memory.buffer-debloat.target` to `duration`.
The default value of the debloat target should be good enough for most cases.

This feature uses past throughout data to predict the time required to consume the remaining
in-flight data. If the predictions are incorrect, the debloating mechanism can fail in one of two ways:
* There will not be enough buffered data to provide full throughput.
* There will be too many buffered in-flight data which will negatively affect the aligned checkpoint barriers propagation time or the unaligned checkpoint size.

If you have a varying load in your Job (i.e. sudden spikes of incoming records, periodically
firing windowed aggregations or joins), you might need to adjust the following settings:

* `taskmanager.network.memory.buffer-debloat.period` - This is the minimum time period between buffer size recalculation. The shorter the period, the faster the reaction time of the debloating mechanism but the higher the CPU overhead for the necessary calculations.

* `taskmanager.network.memory.buffer-debloat.samples` - This adjusts the number of samples over which throughput measurements are averaged out. The frequency of the collected samples can be adjusted via `taskmanager.network.memory.buffer-debloat.period`. The fewer the samples, the faster the reaction time of the debloating mechanism, but a higher chance of a sudden spike or drop of the throughput which can cause the buffer debloating mechanism to miscalculate the optimal amount of in-flight data.

* `taskmanager.network.memory.buffer-debloat.threshold-percentages` - An optimization for preventing frequent buffer size changes (i.e. if the new size is not much different compared to the old size).

Consult the [configuration]({{< ref "docs/deployment/config" >}}#full-taskmanageroptions) documentation for more details and additional parameters.

Here are [metrics]({{< ref "docs/ops/metrics" >}}#io) you can use to monitor the current buffer size:
* `estimatedTimeToConsumerBuffersMs` - total time to consume data from all input channels
* `debloatedBufferSize` - current buffer size
```